coalib/bearlib/languages/documentation/DocumentationComment.py

```diff
@@ -33,3 +38,83 @@ def __init__(self, documentation, language, docstyle,
+        :raises NotImplementedError:
+            When no parsing method is present for the given language and
+            docstyle.
+        """
+        if self.language == "python" and self.docstyle == "default":
+            return self._parse_documentation_with_symbols((":param ", ": "),
+                                                          ":return:")
+        else:
+            raise NotImplementedError(
+                "Documentation parsing for {0.language!r} in {0.docstyle!r}"
+                " has not been implemented yet".format(self))
+
+    def _parse_documentation_with_symbols(self, param_identifiers,
+                                          return_identifiers):
+        """
+        Parses documentation based on parameter and return symbols.
+
+        :param param_identifiers:
+            A tuple of two strings with which a parameter starts
+            and ends.
+        :param return_identifiers:
+            The string with which a return description starts.
+        :return:
+            The list of all the parsed sections of the documentation.
+        """
+        lines = self.documentation.splitlines(keepends=True)
+
+        parse_mode = self.Description
+
+        cur_param = ""
+
+        desc = ""
+        parsed = []
+
+        for line in lines:
+            if line.strip().startswith(param_identifiers[0]):
+                parse_mode = self.Parameter
+                splitted = line[len(param_identifiers[0]):].split(
+                    param_identifiers[1], 1)
+                cur_param = splitted[0].strip()
+                param_desc = ""
+                # For cases where the param description is not on the
+                # same line, but on subsequent lines.
+                try:
+                    param_desc = splitted[1]
+                except IndexError:
+                    pass
+                parsed.append(self.Parameter(name=cur_param, desc=param_desc))
+
+            elif line.strip().startswith(return_identifiers):
+                parse_mode = self.ReturnValue
+                retval_desc = line[len(return_identifiers) + 1:]
+                parsed.append(self.ReturnValue(desc=retval_desc))
+
+            elif parse_mode == self.ReturnValue:
+                retval_desc += line
+                parsed.pop()
+                parsed.append(self.ReturnValue(desc=retval_desc))
+            elif parse_mode == self.Parameter:
+                param_desc += line
+                parsed.pop()
+                parsed.append(self.Parameter(name=cur_param, desc=param_desc))
+            else:
+                desc += line
+                # This is inside a try-except for cases where the list
+                # is empty and has nothing to pop.
+                try:
+                    parsed.pop()
+                except IndexError:
+                    pass
+                parsed.append(self.Description(desc=desc))
+
+        return parsed
```

Comment:
ah right, okay keep this for now, but put the idea using `yield` and regexes into the issue I mentioned above (for improving the algorithm) :+1: